### PR TITLE
Omert/tsk 15387 custom sp reports

### PR DESCRIPTION
**Pull request by marakomer**

### CODE REVIEW

File: scripts/main.go [hunk 1]
```diff
@@ -33,23 +36,56 @@ func main() {
...
 		response_wrapper.HandleResponse("", logger.GetLogs(), action, true)
 		os.Exit(0)
 	}
 	cmdArgs := os.Args[2:]
 
+	// if a mod location was provided,
+	// we clone the repo to the corresponding location mentioned in queryV2PreInvoke in controller
+	if modLocation := os.Getenv(consts.SteampipeReportCustomModLocationEnvVar); modLocation != "" {
+		if err := cloneMod(modLocation); err != nil {
+			log.Errorf("Failed to load the custom Steampipe mod from %s. Please ensure the provided repository is publicly available.", modLocation)
```
Comment (DanielDinsh) [inline scripts/main.go:49]: the user won't see this `log` so the "`Please ensure...`"  is not relevant here no?

File: scripts/main.go [hunk 2]
```diff
@@ -33,23 +36,56 @@ func main() {
...
 		os.Exit(0)
 	}
 
 	response_wrapper.HandleResponse(string(output), logger.GetLogs(), action, false)
 	os.Exit(0)
 }
+
+func cloneMod(repo string) error {
+	modName := extractModName(repo)
+	cmd := exec.Command("git", "clone", repo, filepath.Join(consts.SteampipeBasePath, modName))
```
Comment (DanielDinsh) [inline scripts/main.go:72]: `git` is installed?

File: scripts/main.go [hunk 3]
```diff
@@ -33,23 +36,56 @@ func main() {
...
+		return errors.Wrap(err, cmd.String())
+	}
+	return nil
+}
+
+func extractModName(repo string) string {
+	splitPath := strings.Split(repo, "/")
+	modname := strings.TrimSuffix(splitPath[len(splitPath)-1], ".git")
+	return modname
+}
```
Comment (DanielDinsh) [inline scripts/main.go:91]: [Just a Suggestion] - maybe move this from `main.go` to another file

File: scripts/main.go [hunk 4]
```diff
@@ -33,23 +36,56 @@ func main() {
...
 	os.Exit(0)
 }
+
+func cloneMod(repo string) error {
+	modName := extractModName(repo)
+	cmd := exec.Command("git", "clone", repo, filepath.Join(consts.SteampipeBasePath, modName))
+	if err := cmd.Run(); err != nil {
+		return errors.Wrap(err, cmd.String())
+	}
+	cmd = exec.Command("cd", filepath.Join(consts.SteampipeBasePath, modName))
```
Comment (DanielDinsh) [inline scripts/main.go:76]: extract `filepath.Join(consts.SteampipeBasePath, modName)` and reuse it (it's in use also above)

File: scripts/main.go [hunk 5]
```diff
@@ -33,23 +36,56 @@ func main() {
...
 
 	response_wrapper.HandleResponse(string(output), logger.GetLogs(), action, false)
 	os.Exit(0)
 }
+
+func cloneMod(repo string) error {
+	modName := extractModName(repo)
+	cmd := exec.Command("git", "clone", repo, filepath.Join(consts.SteampipeBasePath, modName))
+	if err := cmd.Run(); err != nil {
+		return errors.Wrap(err, cmd.String())
```
Comment (DanielDinsh) [inline scripts/main.go:74]: this `Wrap` is useful? maybe better -

```go
return errors.Warp(err, "clone mod")
```

File: scripts/main.go [hunk 6]
```diff
@@ -33,23 +36,56 @@ func main() {
...
 		os.Exit(0)
 	}
 	cmdArgs := os.Args[2:]
 
+	// if a mod location was provided,
+	// we clone the repo to the corresponding location mentioned in queryV2PreInvoke in controller
+	if modLocation := os.Getenv(consts.SteampipeReportCustomModLocationEnvVar); modLocation != "" {
+		if err := cloneMod(modLocation); err != nil {
+			log.Errorf("Failed to load the custom Steampipe mod from %s. Please ensure the provided repository is publicly available.", modLocation)
+			response_wrapper.HandleResponse("", logger.GetLogs(), action, true)
```
Comment (DanielDinsh) [inline scripts/main.go:50]: FYI - you are not passing the error to `HandleResponse` so the error message will be the `Generic` one (if I'm not wrong)

**Review by blink-daniel — APPROVED** [review]

**Pull request merged**